driver-async/src/test/unit/com/mongodb/async/client/MongoCollectionSpecification.groovy

```diff
@@ -211,13 +211,16 @@ class MongoCollectionSpecification extends Specification {
 
     def 'should use DistinctOperation correctly'() {
         given:
-        def executor = new TestOperationExecutor([new BsonArray([new BsonString('a')]), new BsonArray([new BsonString('b')])])
-        def filter = new BsonDocument()
+        def asyncCursor = Stub(AsyncBatchCursor) {
+            next(_) >> { args -> args[0].onResult(null, null) }
+        }
+        def executor = new TestOperationExecutor([asyncCursor, asyncCursor])
         def collection = new MongoCollectionImpl(namespace, Document, codecRegistry, readPreference, writeConcern, executor)
+        def filter = new BsonDocument()
         def futureResultCallback = new FutureResultCallback<List<Object>>()
 
         when:
-        collection.distinct('test', filter, futureResultCallback)
+        collection.distinct('test', filter).into([], futureResultCallback)
         futureResultCallback.get()
```

Comment:
I dont think so - because we have these test cases are covered elsewhere, all we care about here is that the DistinctOperation is correctly configured.  OperationIterable and DistinctOperation are tested separately and we have a functional test doing end to end testing.